apis/Google.Cloud.PubSub.V1/docs/index.md

```diff
@@ -82,6 +82,139 @@ restart listening for messages with `StartAsync(...)` again. Due to the expense
 instance, it is recommended that a singleton client per topic is used for the lifetime of the
 application.
 
+## Dependency Injection
+
+Both `PublisherClient` and `SubscriberClient` can be easily integrated with the dependency injection
+container provided by the [Microsoft.Extensions.DependencyInjection](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection/) package.
+
+### PublisherClient
+
+To register a singleton `PublisherClient` instance with default settings in the `IServiceCollection`, use the
+`AddPublisherClient` extension method as shown below:
+
+```csharp
+services.AddPublisherClient(TopicName.FromProjectTopic("projectId", "topicId"));
+```
+
+There is an overload of `AddPublisherClient` method that takes `Action<PublisherClientBuilder>` as a parameter
+and can be used to add the customized `PublisherClient` singleton instance as shown below:
+
+```csharp
+services.AddPublisherClient(builder =>
+{
+    builder.TopicName = TopicName.FromProjectTopic("projectId", "topicId");
+    builder.CredentialsPath = "path/to/credentials.json";
+    // Other settings to customize the client.
+});
+```
+
+Using the `PublisherClient` registered with the dependency injection container is easy. To do so, simply create a constructor for the class in which the `PublisherClient`
```

Comment:
This only works in ASP.NET Core for the types that have automatic integration with DI. An example on how to do this in non ASP.NET application might be useful, maybe more so for subscribers that publishers, but they would be equivalent.

Reply:
I created a sample console app, configured the DI container and the construction injection works just fine as above. Please let me know if I am missing something.

Reply:
You have a controller example which is specific to ASP.NET Core.

`simply create a constructor for the class in which the `PublisherClient` this statement is true only if the type with the constructor is also resolved via DI. In APS.NET Core controllers are resolved via DI automatically, but not on other types of applicatons, i.e. you also need to configure the type with the constructor for DI.
So the better example I think is to show how to resolve a PublisherClient from DI directly, which is the more general example, and then maybe mention that in ASP.NET Core controller's (and other automatically resolved types like middlewares, etc.) that happens automatically. But just maybe.

Also, you have the link to MS's package, but I think it's best to add a link to https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection (to both is fine).

Reply:
Sure Amanda. Will add the link. 
For the non-ASP.NET Core apps, I think it depends on how the DI is configured. For example the below code will work fine in a Console app. However, retrieving a type from the ServiceProvider seems like an anti-pattern to show in docs. Please let me know if you had other thoughts and I can incorporate that. 

```
public class MyService
{
    private readonly PublisherClient _client;
    public MyService(PublisherClient client)
    {
        _client = client;
    }
    
    public void PublishMessage(string message)
    {
    }
}
-------
IServiceCollection services = new ServiceCollection();

// Add services to the container.
services.AddPublisherClient(topicName); 
services.AddSingleton<MyService>();

// Get the ServiceProvider.
var provider = services.BuildServiceProvider();

// Resolve the service and use it.
var service = provider.GetRequiredService<MyService>();
service.PublishMessage(message);

// Or directly resolve the PublisherClient.
var publisher = provider.GetRequiredService<PublisherClient>();
await publisher.PublishAsync(message);
```